core/src/impl/Kokkos_DesulAtomicsConfig.hpp

```diff
@@ -18,6 +18,13 @@
 
 #if KOKKOS_IMPL_ARCH_NVIDIA_GPU < 90
 #define DESUL_CUDA_ARCH_IS_PRE_HOPPER
+#else
+#ifdef KOKKOS_ENABLE_CUDA
+#include <cuda.h>
+#if (CUDA_VERSION >= 12080)
+#define DESUL_HAVE_16BYTE_LOCK_FREE_ATOMICS_DEVICE
+#endif
+#endif
```

Comment:
I'd prefer including `Kokkos_Macros.hpp` as well so we know that `KOKKOS_ENABLE_CUDA` is defined. I would also move the `#ifdef KOKKOS_ENABLE_CUDA` guard to the top of the file (unless we care about other backends being able to use these macros).